### PR TITLE
Developer options menu

### PR DESCRIPTION
**Pull request by stefanceriu**
This PR introduce a Developer Options screen accessible from the app Settings and only on Debug, Nightly and PR builds.
It will also make the existing timeline Developer context menu item only show up in those builds. 

This will be the place where we will introduce toggles for various features while in development and other settings

Screenshot 2023-02-15 at 10 55 50

### CODE REVIEW

File: ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift [hunk 1]
```diff
@@ -30,7 +30,7 @@ struct RoomScreen: View {
             .toolbarBackground(.visible, for: .navigationBar) // Fix the toolbar's background.
             .overlay { loadingIndicator }
             .alert(item: $context.alertInfo) { $0.alert }
-            .sheet(item: $context.debugInfo) { DebugScreen(info: $0) }
+            .sheet(item: $context.debugInfo) { TimelineItemDebugScreen(info: $0) }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift:33]: Random thought: Should we reserve `Screen` for a view with a coordinator, and instead name this `TimelineItemDebugView`?

File: UITests/Sources/DeveloperOptionsScreenScreenUITests.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import ElementX
+import XCTest
+
+class DeveloperOptionsScreenScreenUITests: XCTestCase {
```
Comment (pixlwave) [inline UITests/Sources/DeveloperOptionsScreenScreenUITests.swift:20]: I think you need to update these :)

File: UnitTests/Sources/DeveloperOptionsScreenViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+@testable import ElementX
+
+@MainActor
+class DeveloperOptionsScreenScreenViewModelTests: XCTestCase {
```
Comment (pixlwave) [inline UnitTests/Sources/DeveloperOptionsScreenViewModelTests.swift:22]: And these, but I'm sure CI will yell about these.

File: ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift [hunk 4]
```diff
@@ -229,7 +229,7 @@ class RoomScreenViewModel: RoomScreenViewModelType, RoomScreenViewModelProtocol
             actions.append(.redact)
         }
         
-        var debugActions: [TimelineItemContextMenuAction] = [.viewSource]
+        var debugActions: [TimelineItemContextMenuAction] = ServiceLocator.shared.settings.canShowDeveloperOptions ? [.viewSource] : []
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomScreen/RoomScreenViewModel.swift:232]: 👍

File: ElementX/Sources/Other/EffectsScene/EffectsScene.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import SceneKit
+import SwiftUI
+
+class EffectsScene: SCNScene {
+    private enum Constants {
+        static let confettiSceneName = "ConfettiScene.scn"
+        static let particlesNodeName = "particles"
+    }
+    
+    static func confetti() -> EffectsScene? {
```
Comment (pixlwave) [inline ElementX/Sources/Other/EffectsScene/EffectsScene.swift:26]: Hello old friend 👋

File: ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    var body: some View {
+        Form {
+            Section {
+                Button {
+                    showConfetti = true
+                } label: {
+                    Text("🥳")
+                        .frame(maxWidth: .infinity)
+                }
+            }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift:32]:
```suggestion
            }
            .listRowBackground(Color.element.formRowBackground)
```

File: ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        .background(Color.element.formBackground.ignoresSafeArea())
+        .navigationTitle(ElementL10n.settingsDeveloperOptions)
+        .navigationBarTitleDisplayMode(.inline)
+    }
+    
+    @ViewBuilder
+    private var effectsView: some View {
+        if showConfetti {
+            EffectsView(effect: .confetti)
+                .task { await removeConfettiAfterDelay() }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift:45]: Optional: I always intended to add a way for the view to signal upwards to say that the effect has finished. Can still be time based for now, wouldn't need to hook into SceneKit. Might be a bit tidier.

**Review by pixlwave — APPROVED** [review]
🎉🎉🎉 LGTM 🎉🎉🎉

File: ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift (hunk 7 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/DeveloperOptionsScreen/View/DeveloperOptionsScreenScreen.swift:45]: I looked into it for a bit but couldn't figure it out, also didn't want to spend too much time on it

**Pull request merged**